xonsh/parsers/base.py

```diff
@@ -1973,9 +1977,15 @@ def p_nocloser_many(self, p):
     def p_string_literal(self, p):
         """string_literal : string_tok"""
         p1 = p[1]
-        s = ast.literal_eval(p1.value)
-        cls = ast.Bytes if p1.value.startswith('b') else ast.Str
-        p[0] = cls(s=s, lineno=p1.lineno, col_offset=p1.lexpos)
+        if p1.value.startswith('p'):
+            s = ast.Str(s=ast.literal_eval(p1.value[1:]), lineno=p1.lineno,
+                        col_offset=p1.lexpos)
+            p[0] = xonsh_call('__xonsh_path_literal__', [s],
+                              lineno=p1.lineno, col=p1.lexpos)
```

Comment:
Well, you could have p-strs have a different token than python supported strings.  Then there would be a different parser rule for handling them rather that the rule for handling normal strings.  I think that this is probably the way to go because it will help us be able to support separating out a pure-python parser rules from the extension rules, which is still a goal that we maintain.  

Additionally, doing it this way is slightly faster since you don't have to eval an if-statement that has already effectively been evaluated by the parser.